gpMgmt/bin/gppylib/operations/backup_utils.py

```
@@ -167,6 +232,45 @@ def generate_dump_timestamp(self):
         self.db_date_dir = "%4d%02d%02d" % (year, month, day)
         self.timestamp_object = datetime(year, month, day, hours, minutes, seconds)
 
+    def get_dump_dirs(self):
+        use_dir = self.get_backup_root()
+        dump_path = os.path.join(use_dir, self.dump_dir)
+
+        if not os.path.isdir(dump_path):
+            return []
+
+        initial_list = os.listdir(dump_path)
+        initial_list = fnmatch.filter(initial_list, '[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]')
+
+        dirnames = []
+        for d in initial_list:
+            pth = os.path.join(dump_path, d)
+            if os.path.isdir(pth):
+                dirnames.append(pth)
+
+        if len(dirnames) == 0:
+            return []
+        dirnames = sorted(dirnames, key=lambda x: int(os.path.basename(x)), reverse=True)
+        return dirnames
+
+
+def get_filename_for_content(context, filetype, content, remote_directory=None, host=None):
+    filetype_glob = context.generate_filename(filetype, content=content, directory=remote_directory)
+    if remote_directory:
+        if not host:
+            raise Exception("Must supply name of remote host to check for %s file" % filetype)
+        cmd = Command(name = "Find file of type %s for content %d on host %s" % (filetype, content, host),
+                cmdStr = 'python -c "import glob; print glob.glob(\'%s\')[0]"' % filetype_glob, ctxt = REMOTE, remoteHost = host)
+        cmd.run()
+        if cmd.get_results().rc == 0 and cmd.get_results().stdout:
+            return cmd.get_results().stdout
```

My main concern about this is that we don't check any other types of error. We assume that the only error we'll get is IndexError if glob doesn't anything.

And any other errors are silenced. Very unlikely, but import glob might fail on some other machine since this is a remote command?